manifests/server/etcd.pp

```diff
@@ -120,6 +121,17 @@
   }
 
   if $ensure == 'present' and $manage_members {
+    if defined(Class['k8s']) {
+      $_k8s_cluster_name = $k8s::etcd_cluster_name
+      $_k8s_puppetdb_discovery_tag = $k8s::puppetdb_discovery_tag
+    } else {
+      $_k8s_cluster_name = lookup('k8s::cluster_name', undef, undef, undef)
```

Comment:
why do `lookup('key', undef, undef, undef)`? you also could do `lookup('key')`.

Reply:
Because a lookup without a default value specified (the third of the `undef`s) will error if the lookup fails to find anything.

Reply:
`lookup('key', default_value => undef)` :roll_eyes:

Reply:
I was unaware that Puppet supports brace-less hash creation. Though still using `lookup('key', { default_value => undef })` is definitely a reasonable improvement regardless